src/index.ts

```diff
@@ -0,0 +1,123 @@
+// eslint-disable-next-line import/no-named-default
+import type { default as Oas, Operation } from 'oas';
+
+export const CODE_SAMPLES = 'code-samples';
+export const EXPLORER_ENABLED = 'explorer-enabled';
+export const HEADERS = 'headers';
+export const PROXY_ENABLED = 'proxy-enabled';
+export const SAMPLES_ENABLED = 'samples-enabled';
+export const SAMPLES_LANGUAGES = 'samples-languages';
+export const SEND_DEFAULTS = 'send-defaults';
+export const SIMPLE_MODE = 'simple-mode';
+
+// Make sure you document any changes on here:
+// https://docs.readme.com/docs/openapi-extensions
+export type Extensions = {
+  [CODE_SAMPLES]: {
+    language: string;
+    code: string;
+    name?: string;
+  };
+  [EXPLORER_ENABLED]: boolean;
+  [HEADERS]: { [key: string]: string | number }[];
+  [PROXY_ENABLED]: boolean;
+  [SAMPLES_ENABLED]: boolean; // @deprecated
+  [SAMPLES_LANGUAGES]: string[];
+  [SEND_DEFAULTS]: boolean; // @deprecated
+  [SIMPLE_MODE]: boolean;
+};
+
+export const defaults: Extensions = {
+  [CODE_SAMPLES]: undefined,
+  [EXPLORER_ENABLED]: true,
+  [HEADERS]: undefined,
+  [PROXY_ENABLED]: true,
+  [SAMPLES_ENABLED]: true,
+  [SAMPLES_LANGUAGES]: ['curl', 'node', 'ruby', 'php', 'python'],
+  [SEND_DEFAULTS]: false,
+  [SIMPLE_MODE]: true,
+};
+
+/**
+ * With one of our custom OpenAPI extensions, look for it in either an instance of `oas` or an instance of the
+ * `Operation` class in `oas`.
+ *
+ * Our custom extensions can either be nestled inside of an `x-readme` object or at the root level with an `x-` prefix.
+ *
+ * @see {@link https://npm.im/oas}
+ * @see {@link https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#specificationExtensions}
+ * @see {@link https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#specificationExtensions}
+ * @param extension Specification extension to lookup.
+ * @param oas An instance of the `oas` library class.
+ * @param operation An instsance of the `Operation` class from the `oas` library.
+ * @returns Contents of the extension if found, otherwise the extensions default value.
+ */
+export function getExtension(extension: keyof Extensions, oas: Oas, operation?: Operation) {
+  if (operation) {
+    if (operation.hasExtension('x-readme')) {
```

Comment:
These new `hasExtension` and `getExtension` accessors are part of https://github.com/readmeio/oas/pull/537.